### PR TITLE
audit: improve check for forked Bitbucket repository

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Fixes https://github.com/Homebrew/brew/issues/6681

### CODE REVIEW

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -618,7 +618,7 @@ def audit_bitbucket_repository
 
       new_formula_problem "Uses deprecated mercurial support in Bitbucket" if metadata["scm"] == "hg"
 
-      if metadata["parent"]["full_name"] == "#{user}/#{repo}"
+      if metadata.dig(:parent, :full_name) == "#{user}/#{repo}"
```
Comment (scpeters) [inline Library/Homebrew/dev-cmd/audit.rb]: this is a good fix, but now that I look more closely at this logic, I think the comparison has been wrong. this is comparing the name of the parent repo to the forked repo, which would never be the same. I think we could just see if the `metadata.dig(:parent, :full_name)` is not `nil?` to determine if it's a fork

Comment (jonchang) [reply Library/Homebrew/dev-cmd/audit.rb]: Good call, I like this solution better.

**Comment by MikeMcQuaid** [comment]
Makes sense, thanks @jonchang!

**Comment by zbeekman** [comment]
Thanks Jon!

**Pull request merged**